synapse/handlers/federation.py

```diff
@@ -304,7 +330,7 @@ async def _maybe_backfill_inner(
 
         # We don't want to specify too many extremities as it causes the backfill
         # request URI to be too long.
-        extremities = dict(sorted_extremeties_tuple[:5])
+        extremities = [e.event_id for e in sorted_backfill_points[:5]]
```

Comment:
This just changed from a dict to a list. Is this safe? Presumably we only ever iterated over the dict's keys before?

Reply:
(possibly resolved in other commits?)

Reply:
Only use seems to be passing to FederationEventHandler.backfill, which wants a Collection[str]. So that seems fine.

It eventually makes its way to `event_tuples` here:

https://github.com/matrix-org/synapse/blob/95a038c1069de6c0507eb2c2d9a783c5033a70ec/synapse/federation/transport/client.py#L135-L147

Not sure that `event_tuples` is the best name. But that's not really in scope here.

Reply:
right, sorry, yes, this was subtle, and I should have split it out or made it clearer.

As you note, `extremities` it gets passed into `backfill`, which does nothing but iterate over it. And iterating a `dict` iterates that dict's keys. So, modulo deduplication not happening (which should be ok, because we shouldn't have duplicates here anyway), this is functionally identical.